File: documentation-examples-scala/src/scala/dev/typr/foundations/docs/postgresql/DateTimeTypes.scala
package dev.typr.foundations.docs.postgresql
import dev.typr.scalafoundations.*
import dev.typr.scalafoundations.data.*


import org.postgresql.util.PGInterval
import java.time.{Instant, LocalDate}

@SuppressWarnings(Array("unused"))
object DateTimeTypes:
  //start
  val dateType: PgType[LocalDate] = PgTypes.date
  val timestamptzType: PgType[Instant] = PgTypes.timestamptz
  val intervalType: PgType[PGInterval] = PgTypes.interval
  //stop

File: documentation-examples-java/src/java/dev/typr/foundations/docs/postgresql/DateTimeTypes.java
package dev.typr.foundations.docs.postgresql;

import dev.typr.foundations.PgType;
import dev.typr.foundations.PgTypes;
import java.time.Instant;
import java.time.LocalDate;
import org.postgresql.util.PGInterval;

@SuppressWarnings("unused")
public class DateTimeTypes {
  // start
  PgType<LocalDate> dateType = PgTypes.date;
  PgType<Instant> timestamptzType = PgTypes.timestamptz;
  PgType<PGInterval> intervalType = PgTypes.interval;
  // stop
}
